### PR TITLE
vimPlugins.snacks-nvim: init at 2024-11-07

### PR DESCRIPTION
**Pull request by phrmendes**
## Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/applications/editors/vim/plugins/vim-plugin-names [hunk 1]
```diff
@@ -935,6 +935,7 @@ https://github.com/m4xshen/smartcolumn.nvim/,,
 https://github.com/gorkunov/smartpairs.vim/,,
 https://github.com/ibhagwan/smartyank.nvim/,,
 https://github.com/camspiers/snap/,,
+https://github.com/folke/snacks.nvim,HEAD,
```
Comment (mrcjkb) [inline pkgs/applications/editors/vim/plugins/vim-plugin-names:938]:
```suggestion
https://github.com/folke/snacks.nvim/,HEAD,
```

**Review by GaetanLepage — APPROVED** [review]
LGTM

**Comment by khaneliman** [comment]
Could you squash your commits down, please.

**Review by mrcjkb — CHANGES_REQUESTED** [review]

**Review by mrcjkb — APPROVED** [review]

**Comment by GaetanLepage** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 354276`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>vimPlugins.snacks-nvim</li>
    <li>vimPluginsUpdater</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>vimPlugins.snacks-nvim</li>
    <li>vimPluginsUpdater</li>
  </ul>
</details>

---
### `x86_64-darwin`
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>vimPlugins.snacks-nvim</li>
    <li>vimPluginsUpdater</li>
  </ul>
</details>

---
### `aarch64-darwin`
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>vimPlugins.snacks-nvim</li>
    <li>vimPluginsUpdater</li>
  </ul>
</details>

File: pkgs/applications/editors/vim/plugins/generated.nix [hunk 2]
```diff
@@ -11146,6 +11146,18 @@ final: prev:
...
   };
 
+  snacks-nvim = buildVimPlugin {
+    pname = "snacks.nvim";
+    version = "2024-11-07";
+    src = fetchFromGitHub {
+      owner = "folke";
+      repo = "snacks.nvim";
+      rev = "334895c5bb2ed04f65800abaeb91ccb0487b0f1f";
+      sha256 = "uKAdjRLUtKi6FlE0IHGlnW0Bv2hQhoTRiarvwyU0YaE="
```
Comment (khaneliman) [inline pkgs/applications/editors/vim/plugins/generated.nix:11156]:
```suggestion
      sha256 = "uKAdjRLUtKi6FlE0IHGlnW0Bv2hQhoTRiarvwyU0YaE=";
```
Looks like this got dropped somehow

Comment (phrmendes) [reply pkgs/applications/editors/vim/plugins/generated.nix]: Fixed.

**Review by khaneliman — APPROVED** [review]
LGTM, thanks!

**Pull request merged**